core/src/main/java/org/ehcache/RuntimeConfiguration.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache;
+
+import org.ehcache.config.CacheConfiguration;
+import org.ehcache.config.CacheRuntimeConfiguration;
+import org.ehcache.config.EvictionPrioritizer;
+import org.ehcache.config.EvictionVeto;
+import org.ehcache.config.ResourcePools;
+import org.ehcache.config.ResourceType;
+import org.ehcache.event.CacheEventListener;
+import org.ehcache.event.EventFiring;
+import org.ehcache.event.EventOrdering;
+import org.ehcache.event.EventType;
+import org.ehcache.events.CacheEventNotificationService;
+import org.ehcache.expiry.Expiry;
+import org.ehcache.spi.cache.Store;
+import org.ehcache.spi.service.ServiceConfiguration;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+/**
+ * @author rism
+ */
+public class RuntimeConfiguration<K, V> implements CacheRuntimeConfiguration<K, V>, InternalRuntimeConfiguration<K, V> {
+
+  private final Collection<ServiceConfiguration<?>> serviceConfigurations;
+  private final CacheConfiguration<? super K, ? super V> config;
+  private final Class<K> keyType;
+  private final Class<V> valueType;
+  private final EvictionVeto<? super K, ? super V> evictionVeto;
+  private final EvictionPrioritizer<? super K, ? super V> evictionPrioritizer;
+  private final ClassLoader classLoader;
+  private final Expiry<? super K, ? super V> expiry;
+  private final ResourcePools resourcePools;
+  private final CacheEventNotificationService<K, V> eventNotificationService;
+  private Store<K, V> store;
```

Comment:
Not happy with the `Store` being a field here. From its current use, it could be hidden inside the `CacheEventNotificationService`. And doing that provides a better abstraction for the enabling / disabling of `Store` events.